samples/apps/autogen-studio/autogenstudio/workflowmanager.py

```diff
@@ -93,6 +94,75 @@ def __init__(
         self.sender = None
         self.receiver = None
 
+
+    def _connect_agent_skills(self, skills_module, caller_configs, caller, executor):
+        if caller.llm_config is not False:
+            for skill in caller_configs.get("skills", []):
+                func_name = skill.name
+                func = getattr(skills_module, func_name)
+                # f = caller.register_for_llm(name=func_name, description=skill.description, api_style="function")(func)
+                f = caller.register_for_llm(name=func_name, description=skill.description)(func)
+                executor.register_for_execution(name=func_name)(f)
+
+    def _connect_group_chat_skills(self, skills_module, group_configs, group_chat_manager):
+        agent_configs = group_configs.get("agents", [])
+        group = group_chat_manager.groupchat
+        admin_name = group.admin_name
+        admin_agent = group.agent_by_name(admin_name)
+        count_agents = len(agent_configs)
+        for i in range(count_agents):
+            agent_config = agent_configs[i]
+            agent_name = agent_config["config"].get("name")
+            agent = group.agent_by_name(agent_name)
+            # For some reason I don't yet understand, the group chat manager isn't given the correct name, so
+            # group.agent_by_name(agent_name) fails.
+            if agent == None:
+                # Ideally, every nested group within the group will have it's own unique name, but just in case...
+                groups = [agent for agent in group.agents
+                          if isinstance(agent, ExtendedGroupChatManager) and agent.manager_name == agent_name]
```

Comment:
You'll see later in the code where I added the "agent.manager_name" field to deal with this limitation of AutoGen